phase06/ConsoleApp/SearchEngine/RunProgram.cs

```diff
@@ -0,0 +1,26 @@
+using SampleLibrary;
+
+namespace SearchEngine;
+
+public class RunProgram
+{
+    public void Run(FileReader fileReader)
+    {
+        var documentsDictionary = fileReader.readFiles();
+        var completedInvertedIndex = new InvertedIndex();
+        var invertedIndexMaker = new InvertedIndexMaker(documentsDictionary);
+        completedInvertedIndex = invertedIndexMaker.Make(completedInvertedIndex);
+
+        var io = new IOWorks();
+        var input = io.GetInput();
+
+        var inputProcessor = new InputProcessor();
+        inputProcessor.Process(input);
+
+        var searchInDocs = new SearchInDocs(completedInvertedIndex.Map);
+        HashSet<string> result = searchInDocs.search(inputProcessor.necessary,
+            inputProcessor.optional, inputProcessor.avoided);
+
```

Comment:
why do you have three separate inputs for this function?
it is more readable if you just pass 'inputProcessor' to it as a single input.

Reply:
You're right, but the problem is with the library and should be corrected from the previous phase code.

Reply:
:(